app/controllers/objects_controller.rb

```diff
@@ -12,13 +12,28 @@ def show
     render json: PreservedObject.find_by!(druid: druid).to_json
   end
 
+  # return a specific file from the Moab
+  # GET /objects/:druid/file?type=manifest&filepath=signatureCatalog.xml
+  def file
+    err_msgs = file_params_errors(params)
+    if err_msgs.present?
+      render(plain: "400 Bad Request: #{err_msgs.join('; ')}", status: :bad_request)
+      return
+    end
+
+    file_content = MoabStorageService.retrieve_file(druid, params[:type], params[:filepath], params[:version])
+    render plain: file_content, status: :ok
+  rescue Moab::MoabRuntimeError => e
```

Comment:
Here is another payoff from sul-dlss/moab-versioning/pull/160 - make the superclass of errors for moab-versioning.